jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleConnectionDefinitionBuilder.java

```diff
@@ -0,0 +1,61 @@
+package com.exasol.adapter.dialects.oracle;
+
+import static com.exasol.adapter.AdapterProperties.CONNECTION_NAME_PROPERTY;
+import static com.exasol.adapter.dialects.oracle.OracleProperties.ORACLE_CONNECTION_NAME_PROPERTY;
+import static com.exasol.adapter.dialects.oracle.OracleProperties.ORACLE_IMPORT_PROPERTY;
+
+import java.util.logging.Logger;
+
+import com.exasol.ExaConnectionInformation;
+import com.exasol.adapter.AdapterProperties;
+import com.exasol.adapter.jdbc.BaseConnectionDefinitionBuilder;
+
+/**
+ * This class implements an Oracle-specific connection definition builder
+ */
+public class OracleConnectionDefinitionBuilder extends BaseConnectionDefinitionBuilder {
+    private static final Logger LOGGER = Logger.getLogger(OracleConnectionDefinitionBuilder.class.getName());
+
+    @Override
+    public String buildConnectionDefinition(final AdapterProperties properties,
+            final ExaConnectionInformation exaConnectionInformation) {
+        if (properties.containsKey(ORACLE_IMPORT_PROPERTY)) {
+            return buildImportFromOraConnectionDefinition(properties);
+        } else {
+            return super.buildConnectionDefinition(properties, exaConnectionInformation);
+        }
+    }
+
+    private String buildImportFromOraConnectionDefinition(final AdapterProperties properties) {
+        if (properties.containsKey(ORACLE_CONNECTION_NAME_PROPERTY) && properties.hasUsername()
+                && properties.hasPassword()) {
+            return mixConnectionPropertiesWithOracleConnectionName(properties);
+        } else if (properties.hasConnectionName()) {
+            LOGGER.warning("Exasol named conneciton set in property " + AdapterProperties.CONNECTION_NAME_PROPERTY
+                    + " is ignored during IMPORT FROM ORA.");
+            return buildOracleConnectionDefinition(properties);
+        } else {
+            return buildOracleConnectionDefinition(properties);
```

Comment:
I was not sure whether I should throw an exception here. I was thinking about being more allowing here but since you wrote that mixing named connections and other credentials is prohibited by the frontend, an exception is the right choice here too.